### PR TITLE
Migration of Atom Class w testing

### PR DESCRIPTION
**Pull request by JoshuaSBrown**

### CODE REVIEW

**Comment by junghans** [comment]
Sorry for my ignorance, but where is the class coming from? And shouldn't it be derived from `votca::csg::bead`? (Or `bead` from the new `atom`)

File: include/votca/xtp/atom.h [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        _weight(stencil->getWeight()),
+        _pos(stencil->getPos()),
+        _bPos(true),
+        _hasQM(stencil->HasQMPart()),
+        _qmId(stencil->getQMId()),
+        _qmPos(stencil->getQMPos()),
+        _element(stencil->getElement()) {}
+
+  Atom() {};
+  ~Atom() { _Q.clear(); }
```
Comment (MrTheodor) [inline include/votca/xtp/atom.h]: This is not required

File: include/votca/xtp/atom.h [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  void setPos(const votca::tools::vec &r);
+  /**
+   * direct access (read/write) to the position of the atom
+   * \return reference to position
+   */
+  votca::tools::vec &Pos() { return _pos; }
+  /** does this configuration store positions? */
+  bool HasPos() { return _bPos; }
+  /** dose the bead store a position */
+  void HasPos(bool b);
```
Comment (MrTheodor) [inline include/votca/xtp/atom.h]: camelCase or CamelCase ?

File: src/libxtp/espfit.cc [hunk 3]
```diff
@@ -81,7 +81,11 @@ void Espfit::Fit2Density(std::vector< QMAtom* >& _atomlist, ub::matrix<double> &
     CTP_LOG(ctp::logDEBUG, *_log) << ctp::TimeStamp() << " Calculating ESP at CHELPG grid points"  << flush;
     //boost::progress_display show_progress( _grid.getsize() );
     #pragma omp parallel for
+<<<<<<< HEAD
     for ( unsigned i = 0 ; i < _grid.getsize(); i++){
+=======
+    for ( unsigned int i = 0 ; i < _grid.getsize(); i++){
+>>>>>>> fc32ec7d18c290b005462342e604b450817c3563
```
Comment (junghans) [inline src/libxtp/espfit.cc]: @JoshuaSBrown : these conflict markers need to go!

File: src/libxtp/espfit.cc [hunk 4]
```diff
@@ -112,7 +116,10 @@ void Espfit::Fit2Density(std::vector< QMAtom* >& _atomlist, ub::matrix<double> &
 ub::vector<double> Espfit::EvalNuclearPotential(std::vector< QMAtom* >& _atoms, Grid _grid) {
     ub::vector<double> _NucPatGrid = ub::zero_vector<double>(_grid.getsize());
 
+<<<<<<< HEAD
    
+=======
+>>>>>>> fc32ec7d18c290b005462342e604b450817c3563
```
Comment (junghans) [inline src/libxtp/espfit.cc]: Same!

Comment (JoshuaSBrown) [reply src/libxtp/espfit.cc]: Yes, yes. You are highlighting my inexperience with git. :)

**Pull request merged**